cql3/statements/alter_keyspace_statement.cc

```diff
@@ -82,20 +82,68 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
 #endif
 }
 
+#include "locator/load_sketch.hh"
+#include "service/topology_guard.hh"
+#include "service/topology_mutation.hh"
+#include "locator/tablet_replication_strategy.hh"
+
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
 cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_processor& qp, api::timestamp_type ts) const {
     try {
         auto old_ksm = qp.db().find_keyspace(_name).metadata();
-        const auto& tm = *qp.proxy().get_token_metadata_ptr();
+        const auto& tm = qp.proxy().get_token_metadata_ptr();
+        auto m = service::prepare_keyspace_update_announcement(qp.db().real_database(), _attrs->as_ks_metadata_update(old_ksm, *tm), ts);
+
+        auto&& replication_strategy = qp.db().find_keyspace(_name).get_replication_strategy();
+        if (replication_strategy.uses_tablets()) {
+            const auto& dcs = tm->get_topology().get_datacenters();
+            const auto& tables = old_ksm->tables()[0]->id();
+
+            if (const locator::tablet_aware_replication_strategy* tablet_rs = qp.db().find_keyspace(_name).get_replication_strategy().maybe_as_tablet_aware()) {
+                // TODO: get proper schema_ptr
+                auto map = tablet_rs->reallocate_tablets_for_existing_table(schema_ptr{}, tm).get0();
+                for (auto const& ks_table : old_ksm->tables()) {
+                    m.emplace_back(replica::tablet_map_to_mutation(map, ks_table->id(), ks_table->keypace_name(),
+                                                                   ks_table->cf_name(), ts).get0());
+                }
+            }
+
+            if (not topology.is_busy()) // check is topology state machine is free to process this request?
```

Comment:
is_busy() is not the right check for this. is_busy() means you cannot start handling a request, but you can still queue a request. What we should check is whether there is room for queuing a request (we currently support only 1 outstanding request).